include/mbedtls/platform_util.h

```diff
@@ -167,7 +167,28 @@ MBEDTLS_DEPRECATED typedef int mbedtls_deprecated_numeric_constant_t;
  * \param len   Length of the buffer in bytes
  *
  */
+#if defined(MBEDTLS_PLATFORM_ZEROIZE_CHECK_UNSAFE)
```

Comment:
```suggestion
#if defined(MBEDTLS_PLATFORM_ZEROIZE_CHECK_UNSAFE) && defined(MBEDTLS_TEST_HOOKS)
```

Maybe this can help reduce the risk of this accidentally getting enabled?

Reply:
Well, `MBEDTLS_TEST_HOOKS`, is a documented config option in `mbedtls_config.h`. It has to be manually enabled in `all.sh` if wanted. The name has got "TEST" in the name, indicating that it might not necessarily be wanted in production.

`MBEDTLS_PLATFORM_ZEROIZE_CHECK_UNSAFE` is only documented just under where it is used (to placate check-names, otherwise I would have not put it in). It needs to be manually enabled when wanted, which it is in the test in `all.sh`. The name has got "UNSAFE" in it, indicating that it really shouldn't be used in production. It's not in `mbedtls_config.h`, so someone running through enabling everything without careful reading is not going to get it.

So: this would seem to be the equivalent of needing to turn two keys before launching a missile, rather than just one.

I'm happy to add it if you think it really necessary, but I think `MBEDTLS_TEST_HOOKS` is more likely to be accidentally enabled than `MBEDTLS_PLATFORM_ZEROIZE_CHECK_UNSAFE` is, so I don't know if it's actually worth it.

Reply:
> So: this would seem to be the equivalent of needing to turn two keys before launching a missile, rather than just one.

Yes, which is why our policy is that it takes more than doing some key turning to launch a missile. `MBEDTLS_TEST_HOOKS` by itself must not enable any insecure behavior, only the possibility of injecting some code (not from the library) which itself may be insecure.